### PR TITLE
Is cutadapt paired upstream ready

### PR DESCRIPTION
**Pull request by IanSudbery**
Edited the cutadapt preprocessor to include options to perform the processing in a pariedend manner. The advantage of this over single end processing followed by reconcile is the amount of memory that reconcile requires.

### CODE REVIEW

**Comment by IanSudbery** [comment]
I'd appreaciate a second pair of eyes on this.

File: CGATPipelines/PipelinePreprocess.py [hunk 1]
```diff
@@ -512,23 +516,50 @@ def build(self, infiles, outfiles, output_prefix):
...
+                format = ""
+
+            untrimmed_output1, untrimmed_output2 = \
+                [i.replace(".fast", "_untrimmed.fast")
+                 for i in infiles]
+
+            if untrimmed:
+                processing_options += \
+                    "--untrimmed-output=%(untrimmed_output1)s" \
+                    "--untrimmed-paired=%(untrimmed_output2)s" % locals()
```
Comment (TomSmithCGAT) [inline CGATPipelines/PipelinePreprocess.py]: Should line 537 be `"--untrimmed-output-paired=%(untrimmed_output2)s" % locals()`?

`cutadapt --help|grep untrimmed`
    --no-trim           Match and redirect reads to output/untrimmed-output as
    --discard-untrimmed, --trimmed-only
    --untrimmed-output=FILE
    --untrimmed-paired-output=FILE
                        together with --untrimmed-output when trimming paired-

`cutadapt --version`
1.9.1

**Comment by IanSudbery** [comment]
Cheers Tom. Thats exactly why I asked for eyes on! We'd had the paired output working for months, but hadn't intergrated it with the untrimmed output options

**Comment by TomSmithCGAT** [comment]
No worries. All looks good to me. Agree this is much better than using reconcile.

**Pull request merged**